File: Benchmarking/src/gen597.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: [s0]
    var p1: [s0]
  }
  func f15(_ v0: s0) -> s0 {
    let v4: [s0] = [v0, v0, v0, v0, v0]
    let v1: [s0] = [v0, v0, v0, v0, v0, v0, v0]
    let v6: s0 = v4[1]
    var v19: [s0] = v1
    let v23: s0 = v19[0]
    v19[4] = v6
    var v61: s0 = v23
    return v61
  }
  @inline(never)
  func f0(_ v0: s0, _ v1: s1, _ v2: Double) -> Double {
    let v9: s0 = f15(v0)
    let v13: [[Double]] = v9.p0
    let v57: [Double] = v13[1]
    let v87: Double = v57[0]
    return v87
  }
  func benchmark() {
    let v0: s0 = s0(p0: [[0.0], [1.0], [2.0]], p1: [[3.0]])
    let v1: s1 = s1(p0: [s0(p0: [[4.0], [5.0], [6.0]], p1: [[7.0]])], p1: [s0(p0: [[8.0], [9.0], [10.0]], p1: [[11.0]]), s0(p0: [[12.0], [13.0], [14.0]], p1: [[15.0]]), s0(p0: [[16.0], [17.0], [18.0]], p1: [[19.0]])])
    let v2: Double = 20.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 20.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen597.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    std::vector<s0> p0;
    std::vector<s0> p1;
    s1(std::vector<s0> v0, std::vector<s0> v1): p0(v0), p1(v1) { }
  };
  s0 f15(const s0 &v0) {
    const std::vector<s0> v4 { v0, v0, v0, v0, v0 };
    const std::vector<s0> v1 { v0, v0, v0, v0, v0, v0, v0 };
    const s0 v6 = v4[1];
    std::vector<s0> v19 = v1;
    const s0 v23 = v19[0];
    v19[4] = v6;
    s0 v61 = v23;
    return v61;
  }
  __attribute__((noinline))
  double f0(const s0 &v0, const s1 &v1, const double &v2) {
    const s0 v9 = f15(v0);
    const std::vector<std::vector<double>> v13 = v9.p0;
    const std::vector<double> v57 = v13[1];
    const double v87 = v57[0];
    return v87;
  }
  int main() {
    s0 v0({ { { 0.0 }, { 1.0 }, { 2.0 } }, { { 3.0 } } });
    s1 v1({ { { { { 4.0 }, { 5.0 }, { 6.0 } }, { { 7.0 } } } }, { { { { 8.0 }, { 9.0 }, { 10.0 } }, { { 11.0 } } }, { { { 12.0 }, { 13.0 }, { 14.0 } }, { { 15.0 } } }, { { { 16.0 }, { 17.0 }, { 18.0 } }, { { 19.0 } } } } });
    double v2(20.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }
